From 0b00dc9f4293411537f21885195d30a3aeebe821 Mon Sep 17 00:00:00 2001
From: Ajay Singh Dahiya <asinghda@qti.qualcomm.com>
Date: Mon, 14 Jul 2025 11:43:32 +0530
Subject: [PATCH 27/40] qcacld-3.0: Add function pointer for HAPS event
 handling

Added a function pointer during HAPS attachment to support
HTT message processing.

Change-Id: If923445d503e124f98b5a728eda057cee9f574cc
CRs-Fixed: 4216009
---
 components/dp/core/inc/wlan_dp_haps.h |  4 ----
 components/dp/core/src/wlan_dp_haps.c | 12 +++++++++---
 2 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/components/dp/core/inc/wlan_dp_haps.h b/components/dp/core/inc/wlan_dp_haps.h
index b5ec427cde..312688acf9 100644
--- a/components/dp/core/inc/wlan_dp_haps.h
+++ b/components/dp/core/inc/wlan_dp_haps.h
@@ -33,10 +33,6 @@ void dp_vdev_haps_attach(struct cdp_soc *psoc, struct wlan_dp_intf *dp_intf,
 
 void dp_vdev_haps_detach(struct wlan_dp_intf *dp_intf);
 
-void dp_haps_handle_ind(ol_osif_vdev_handle osif_vdev, haps_state new_state,
-			qdf_ktime_t timeout, bool is_one_shot,
-			bool is_direct_reg_write);
-
 void dp_haps_init(struct wlan_objmgr_psoc *psoc);
 
 QDF_STATUS dp_print_haps_stats(struct dp_soc *soc);
diff --git a/components/dp/core/src/wlan_dp_haps.c b/components/dp/core/src/wlan_dp_haps.c
index 36fe7dfd2d..f145ad1364 100644
--- a/components/dp/core/src/wlan_dp_haps.c
+++ b/components/dp/core/src/wlan_dp_haps.c
@@ -138,9 +138,10 @@ void dp_haps_update_state(struct dp_haps *haps_ctx, haps_state new_state,
  *
  * Returns: none
  */
-void dp_haps_handle_ind(ol_osif_vdev_handle osif_vdev, haps_state new_state,
-			qdf_ktime_t time_rcvd, bool is_one_shot,
-			bool is_direct_reg_write)
+static void
+dp_haps_handle_ind(ol_osif_vdev_handle osif_vdev, haps_state new_state,
+		   qdf_ktime_t time_rcvd, bool is_one_shot,
+		   bool is_direct_reg_write)
 {
 	struct dp_haps *haps_ctx = dp_get_haps_ctx_from_vdev(osif_vdev);
 	uint64_t curr_time_us = qdf_get_log_timestamp_usecs();
@@ -305,6 +306,10 @@ void dp_clear_haps_stats(struct dp_soc *soc)
 	}
 }
 
+static struct cdp_haps_ops dp_ops_haps = {
+	.haps_handle_ind = dp_haps_handle_ind,
+};
+
 /**
  * dp_vdev_haps_attach() - Attach the HAPS
  * @psoc: Datapath global soc handle
@@ -346,6 +351,7 @@ void dp_vdev_haps_attach(struct cdp_soc *psoc, struct wlan_dp_intf *dp_intf,
 			 QDF_HRTIMER_MODE_REL,
 			 QDF_CONTEXT_TASKLET);
 
+	soc->cdp_soc.ops->haps_ops = &dp_ops_haps;
 	dp_info("HAPS is successfully attached for vdev:%u\n", vdev_id);
 }
 
-- 
2.34.1

